methods.py

```diff
@@ -617,7 +617,11 @@ def detect_darwin_sdk_path(platform, env):
             raise
 
 def get_compiler_version(env):
-    version = decode_utf8(subprocess.check_output([env['CXX'], '--version']).strip())
+    # Not using this method on clang because it returns 4.2.1 # https://reviews.llvm.org/D56803
+    if using_gcc(env):
+        version = decode_utf8(subprocess.check_output([env['CXX'], '-dumpversion']).strip())
+    else:
+        version = decode_utf8(subprocess.check_output([env['CXX'], '--version']).strip())
     match = re.search('[0-9][0-9.]*', version)
```

Comment:
Alternatively, escaping the dot in the regex should be enough to fix it, no?

Reply:
It doesn't hurt to use `-dumpversion` for GCC though, which is indeed a better way to retrieve the version.

Reply:
I'm not good at regex, give me a regex to test.

Though I think this is a cleaner way to get the version.

Reply:
`.` matches any character in a regex, so to match the actual `.` character, it should be escaped: `[0-9][0-9\.]*`.

This would match:
- `[0-9]`: Exactly one character comprised between 0 and 9
- `[0-9\.]*`: Zero or more characters comprised between 0 and 9 or `.`

So anything that starts with a number and is then followed by more numbers and/or dots.

Reply:
Actually I just checked and the `.` doesn't need to be escaped here, so it's fine as is.

Reply:
Just found this https://stackoverflow.com/questions/45168516/gcc-7-1-1-on-fedora-26-dumpversion-now-only-includes-major-version-by-default
Searched the scons scripts and and I think we check only for the major version for now, so it should be ok. But if in the future we check the minors it won't work this way.